SCClassLibrary/Common/Files/SoundFile.sc

```diff
@@ -70,6 +71,38 @@ SoundFile {
 		^res
 	}
 
+    *writeArray {|pathName, array, headerFormat, sampleRate, sampleFormat|
+        var file = SoundFile.new;
+
+        if (array.rank > 2) {
+            Error(
+                "Cannot write an array with nested channels."
+                "The array should be formatted as: "
+                "[  [channel 1 samples...], [channel 2 samples...], ...]"
+            ).throw
+        };
+
+        // Assign if not nil.
+        headerFormat !? file.headerFormat_(_);
+        sampleFormat !? file.sampleFormat_(_);
+        sampleRate !? file.sampleRate_(_);
+
+        file.numChannels_(if(array.rank <= 1, { 1 }, { array.shape[0] }));
+
+        if(file.openWrite(pathName).not) {
+            Error("Could not open file at path:" + pathName).throw
+        };
+
+        // Interlace the samples and write, always close the file.
+        // Format will be: L1, R1, C1, L2, R2, C2 ...
+        protect {
+            file.writeData(
+                FloatArray.newFrom(array.lace(array.shape.reduce('*')).asFloat)
```

Comment:
stylistic (using variable name to convey intent):

```
var size = array.shape.reduce('*');
var flattenedArray = array.lace(size); // .lace is faster than .flop.flat here
var data = FloatArray.newFrom(flattenedArray.asFloat);
file.writeData(data)
```
(if you like, you could use `product` instead of `reduce('*')`, but this is a matter of taste.